src/main/java/com/google/api/codegen/go/GoGapicContext.java

```diff
@@ -101,6 +101,34 @@ public GoGapicContext(Model model, ApiConfig apiConfig) {
     this.goCommon = new GoContextCommon();
   }
 
+  /**
+   * Returns the Go package name.
+   *
+   * Retrieved by splitting the package string on `/` and returning the second to last string if
+   * possible, and otherwise the last string in the resulting array.
+   */
+  public String getPackageName() {
```

Comment:
No, unfortunately, not consistently. I think this will be fine temporarily. I think the only way around it is to add a field to the config, but it may also be a go only issue.
Will follow up in this.